lib/mutant/mutation.rb

```diff
@@ -40,11 +40,11 @@ def original_source
 
     # Test if mutation is killed by test reports
     #
-    # @param [Array<Report::Test>] test_reports
+    # @param [Result::Test] test_result
     #
     # @return [Boolean]
-    def self.success?(test_result)
-      self::TEST_PASS_SUCCESS.equal?(test_result.passed)
+    def success?(test_result)
```

Comment:
It is de-facto using instance state `self.class::` for looking up a constant relative to the instances class.

That we factored this out in a constant is just implementation detail and should not leak into an ugly public interface: https://github.com/mbj/mutant/pull/514/files#diff-e4388dd312e80786e0cc9aaa0272e00bL217

An ugly public interface that was uneasy to spec because it would have required many hops in method expectation chains to do so.

Also conceptually its the mutation *instances* decision if a given test result kills that mutation. Its just by accident we do not *yet* use other information than the `Result::Test#pass`. In other experiments I want to base of this code (some of these are not in this repo) also the time spend on the mutation is relevant relative to the subjects (reachable from the mutation instance) configuration.